transcom/mymove
# B-21049 INT

## B-21049

## Summary

In DocumentViewer, create an exception handler to be passed into Content and into the react-file-viewer:
* Display a message where the content should be, "This document is not yet available for viewing. Please try again in a moment."
* _Disable the download link_ (Left the link, conflicted with integration tests)
* Hide the react-file-viewer’s “Loading NaN%” message
it uses CSS class “pdf-loading”

An update in react-file-viewer was needed to make use of the provided onError parameter.

## Verification Steps for the Author

These are to be checked by the author.


## Verification Steps for Reviewers

These are to be checked by a reviewer.


### Setup to Run the Code

- [Instructions for starting storybook](https://transcom.github.io/mymove-docs/docs/frontend/setup/storybook)
- [Instructions for starting the MilMove application](https://transcom.github.io/mymove-docs/docs/getting-started/application-setup/)
- [Instructions for running tests](https://transcom.github.io/mymove-docs/docs/getting-started/development/testing)

### How to test

1. Access the office app
2. Login as a TOO
3. Search for and select a customer
4. Supporting Documents
5. Upload a PDF or image file
6. To simulate the AWS S3 delay, delete the newest file from mymove/tmp/storage/user/*/uploads/
7. Refresh the Supporting Documents page
8. see the File Not Found message and no download button

### Frontend


### Backend


### Database

#### Any new migrations/schema changes:


## Screenshots
![Screenshot 2024-11-04 at 10 03 39 AM](https://github.com/user-attachments/assets/28627ebd-0eaa-4105-98ca-5a4448c61e23)
